builtin/providers/vcd/resource_vcd_firewall_rules.go

```diff
@@ -0,0 +1,241 @@
+package vcd
+
+import (
+	"bytes"
+	"fmt"
+	"github.com/hashicorp/terraform/helper/hashcode"
+	"github.com/hashicorp/terraform/helper/schema"
+	"github.com/hmrc/vmware-govcd"
+	types "github.com/hmrc/vmware-govcd/types/v56"
+	"strings"
+)
+
+func resourceVcdFirewallRules() *schema.Resource {
```

Comment:
We have found in other providers that going with a singular resource definition (so `vcd_firewall_rule` instead of `vcd_firewall_rules`) allows for much more flexibility in Terraform config. For instance, it enables one team to produce a module of "base rules" to which other teams can add app-specific rules. When implemented as a set like this, all rules have to be known at one time, and it's impossible to compose rules.

Do you think you could look into reworking this as a single `vcd_firewall_rule` resource?